endpoints/cookie_sync.go

```diff
@@ -88,6 +93,14 @@ func (deps *cookieSyncDeps) Endpoint(w http.ResponseWriter, r *http.Request, _ h
 		http.Error(w, "gdpr_consent is required if gdpr=1", http.StatusBadRequest)
 		return
 	}
+	// If GDPR is ambiguous, lets untangle it here.
+	if parsedReq.GDPR == nil {
+		if deps.gDPR.UsersyncIfAmbiguous {
+			parsedReq.GDPR = &zero
+		} else {
+			parsedReq.GDPR = &one
```

Comment:
This may get us into a position where `*parsedReq.GDPR == 1` but `gdpr_consent == ""`, which throws an error in the above block. This is probably ok if `permissions.HostCookiesAllowed` (via `filterForGDPR`) handles this case without error, otherwise in the `UsersyncIfAmbiguous == false` case we could set `*parsedReq.GDPR == 0` and ``*parsedReq.Bidders == nil`.

Reply:
Thats why I put this just below the above block. :)  Thought it best to avoid `nil` as either we would have to push the config option and the logic deeper into the code, or risk the original issue creeping back. We already have lots of checks `if consent == ""` so I think we are on solid ground for the case of `gdpr: 1` and no consent string, but it is a good point.

Reply:
I think @hhhjort has the right idea here. Or at least: the block above returns a 400. I don't think that makes sense if gdpr/consent are undefined, because it's a perfectly valid request.